### ISSUE TITLE
Callout: Get a crash due to Uncaught Invariant Violation

### ISSUE THREAD

**Issue by jspurlin** [OfficeDev/office-ui-fabric-react#10701]
In word online I'm seeing an issue when bringing up a callout (with a point target) that's causing our app to crash.

I believe was introduced by: #10663 ,fyi @dzearing and @JasonGore 

This is a High impact bug

Below is the console exception:

Uncaught Invariant Violation: Argument appears to not be a ReactComponent. Keys: x,y
    at eval (webpack-internal:///../../common/temp/node_modules/.office.pkgs.visualstudio.com/react-dom/16.10.1_react@16.10.1/node_modules/react-dom/cjs/react-dom.development.js:27025:32)
    at findHostInstanceWithWarning (webpack-internal:///../../common/temp/node_modules/.office.pkgs.visualstudio.com/react-dom/16.10.1_react@16.10.1/node_modules/react-dom/cjs/react-dom.development.js:27028:11)
    at findDOMNode (webpack-internal:///../../common/temp/node_modules/.office.pkgs.visualstudio.com/react-dom/16.10.1_react@16.10.1/node_modules/react-dom/cjs/react-dom.development.js:27722:14)
    at getWindow (webpack-internal:///../../common/temp/node_modules/.office.pkgs.visualstudio.com/@uifabric/utilities/7.3.0_1d11bb06501bc9714a7757ddc6fa6f6b/node_modules/@uifabric/utilities/lib/dom/getWindow.js:31:123)
    at Async.requestAnimationFrame (webpack-internal:///../../common/temp/node_modules/.office.pkgs.visualstudio.com/@uifabric/utilities/7.3.0_1d11bb06501bc9714a7757ddc6fa6f6b/node_modules/@uifabric/utilities/lib/Async.js:389:83)
    at CalloutContentBase._updateAsyncPosition (webpack-internal:///../../common/temp/node_modules/.office.pkgs.visualstudio.com/office-ui-fabric-react/7.42.0_1d11bb06501bc9714a7757ddc6fa6f6b/node_modules/office-ui-fabric-react/lib/components/Callout/CalloutContent.base.js:234:21)
    at CalloutContentBase._this._onComponentDidMount (webpack-internal:///../../common/temp/node_modules/.office.pkgs.visualstudio.com/office-ui-fabric-react/7.42.0_1d11bb06501bc9714a7757ddc6fa6f6b/node_modules/office-ui-fabric-react/lib/components/Callout/CalloutContent.base.js:81:19)
    at CalloutContentBase.componentDidMount (webpack-internal:///../../common/temp/node_modules/.office.pkgs.visualstudio.com/office-ui-fabric-react/7.42.0_1d11bb06501bc9714a7757ddc6fa6f6b/node_modules/office-ui-fabric-react/lib/components/Callout/CalloutContent.base.js:162:18)
    at commitLifeCycles (webpack-internal:///../../common/temp/node_modules/.office.pkgs.visualstudio.com/react-dom/16.10.1_react@16.10.1/node_modules/react-dom/cjs/react-dom.development.js:21964:22)
    at commitLayoutEffects (webpack-internal:///../../common/temp/node_modules/.office.pkgs.visualstudio.com/react-dom/16.10.1_react@16.10.1/node_modules/react-dom/cjs/react-dom.development.js:25288:7)

**Comment by JasonGore** [comment]
#10696 was merged to fix similar Callout issues. Do you know if this is still an issue after the merge of #10696?

**Comment by dzearing** [comment]
This is resolved. Please patch with 7.43.0.

**Comment by dzearing** [comment]
@whykushal93 is validating, once we get confirmation that the issue has been resolved, we'll close

**Comment by danmarshall** [comment]
Hello, I also get a crash for my scenario with a DropDown that I try to call `.focus()` upon after it is mounted.

```
this.dropdownRef.current!.focus(true);  // current is null in v6.205.0
```

I am getting this error in 6.205.0, it works in 6.204.4

**Comment by dzearing** [comment]
@danmarshall not reproing;

https://codesandbox.io/s/dropdown-focus-vvvro

**Comment by danmarshall** [comment]
Thanks @dzearing - I will try to create a minimal repro. Until then, can you visit https://danmarshall.github.io/SandDance/app/ and click on "Latitude"

![image](https://user-images.githubusercontent.com/11507384/66361086-495d5200-e932-11e9-8d71-e789ae3435ed.png)

**Comment by danmarshall** [comment]
For comparison, on our production app https://sanddance.js.org/app/ I am using 6.204.4, getting desired behavior:
![image](https://user-images.githubusercontent.com/11507384/66361572-22a01b00-e934-11e9-844c-20e096194394.png)

**Comment by danmarshall** [comment]
Ok, I have the minimal repro:
https://codesandbox.io/embed/dropdown-focus-zgujc
Thanks,
Dan

**Comment by dzearing** [comment]
Awesome, thank you @danmarshall ! I found it.

Dropdown uses the withResponsiveMode decorator. The decorator wants to hold up rendering content until it knows how big things are. This delay is causing the parent to be mounted, and then the child to be mounted.

I'll figure out a better solution.

**Comment by dzearing** [comment]
Alright, PR out plus test, and will get that merged into 6.0 immediately after. Sorry for the trouble.

**Comment by danmarshall** [comment]
Hey, no trouble, you guys rock!

**Comment by dzearing** [comment]
@danmarshall your site is awesome, btw!

**Comment by danmarshall** [comment]
The office-ui-fabric-react makes it shine!

**Comment by whykushal93** [comment]
@dzearing we see a similar issue for the `Dialog` componnent. The Dialog Inside componentDidMount tries to focus an inner element. But, it does not seem to work because of the recent change to withResponsiveMode decorator. 

Do you think the PR have open will also fix that issue ? 

cc: @jspurlin

**Comment by dzearing** [comment]
This is now fixed in v7.47.0. @danmarshall @whykushal93 @lorejoh12 just a heads up here, this is published now, and I've cherrypicked into v6 as well which is in the queue publishing.

**Comment by dzearing** [comment]
Verified this is resolved.
